gui/wxpython/datacatalog/tree.py

```diff
@@ -294,6 +294,15 @@ def __init__(
             self.grassdatabases.append(currentDB)
             self._saveGrassDBs()
 
+        self.ReloadTreeItems()
+
+        # If demolocation, show the map world layer in the Map Display
+        location = gisenv()['LOCATION_NAME']
+        mapset = gisenv()['MAPSET']
+        if (location == "world_latlong_wgs84"
+        and mapset == "PERMANENT"):
```

Comment:
A dedicated function to decide if you are in the demo location might be a good idea. Something like `current_mapset_is_in_demolocation()` in `grass.grassdb`. BTW, `grass.grassdb` may need to eventually have functions from `startup.utils`, but that's unrelated to this PR.